src/utp_stream.cpp

```diff
@@ -261,8 +407,9 @@ struct packet
 struct utp_socket_impl
 {
 	utp_socket_impl(std::uint16_t recv_id, std::uint16_t send_id
-		, void* userdata, utp_socket_manager* sm)
+		, void* userdata, utp_socket_manager* sm, packet_pool* pp)
 		: m_sm(sm)
+		, m_packet_pool(pp)
```

Comment:
what would you think about making the packet allocation and freeing functions available via the ``utp_socket_manager`` instead? that way each individual socket would not need to carry the extra pointer